annotated_test.go

```diff
@@ -1448,10 +1448,40 @@ func TestAnnotate(t *testing.T) {
 
 		err := app.Err()
 		require.Error(t, err)
-		assert.Contains(t, err.Error(), "fx.Out structs cannot be annotated")
+		assert.Contains(t, err.Error(), "fx.Out structs cannot be annotated with either of fx.ResultTags, fx.As or fx.From")
 	})
 
-	t.Run("annotate a fx.In", func(t *testing.T) {
+	t.Run("annotate a fx.Out with As", func(t *testing.T) {
```

Comment:
Depending on what you do with my first comment, you might want to add more test cases. Specifically, with the current implementation, you're missing test cases using fx.From with fx.Out and fx.As with fx.In. If you decide to let users be able to use those cases, you don't need to add them to tests.